cinex/PageModels/MainPageModel.cs

```diff
@@ -0,0 +1,213 @@
+﻿using cinex.Models;
+using cinex.Services;
+using FreshMvvm;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+
+namespace cinex.PageModels
+{
+    [PropertyChanged.AddINotifyPropertyChangedInterface]
+    public class MainPageModel : FreshBasePageModel
+    {
+        IAppService _appService;
+
+        private int NextPage { get; set; } = 1;
+
+        private int TotalPages { get; set; } = 1;
+
+        public bool IsLoading { get; set; }
+
+        public string Title { get; set; }
+
+        public ObservableCollection<UpcomingModel> Upcomings { get; }
+
+        public List<UpcomingModel> OriginalUpcomings { get; set; }
+
+        public List<GenreModel> Genres { get; }
+
+        public bool ShowLoadMore { get; set; }
+
+        private string _searchString;
+        public string SearchString
+        {
+            get => _searchString;
+            set
+            {
+                if (!EqualityComparer<string>.Default.Equals(_searchString, value))
+                {
+                    _searchString = value;
+                    FilterUpcomings();
+                }
+
+                ShowLoadMore = string.IsNullOrEmpty(_searchString);
+            }
+        }
+
+        private UpcomingModel _selectedItem;
+        public UpcomingModel SelectedItem
+        {
+            get => _selectedItem;
+            set
+            {
+                _selectedItem = value;
+                if (_selectedItem != null)
+                {
+                    ShowUpcomingDetailsCommand.Execute(_selectedItem);
+                }
+            }
+        }
+
+        public MainPageModel(IAppService appService)
+        {
+            Title = "CineX - Upcomings";
+            OriginalUpcomings = new List<UpcomingModel>();
+            Upcomings = new ObservableCollection<UpcomingModel>();
+            Genres = new List<GenreModel>();
+            _appService = appService;
+        }
+
+        public override void Init(object initData)
+        {
+            base.Init(initData);
+        }
+
+        protected override async void ViewIsAppearing(object sender, EventArgs e)
+        {
+            if (Upcomings.Count == 0)
+            {
+                this.IsLoading = true;
+
+                await LoadGenres();
+                await LoadUpcomings();
+
+                this.IsLoading = false;
+            }
+
+            SelectedItem = null;
+        }
+
+        public Command LoadMoreCommand
+        {
+            get
+            {
+                return new Command(async () =>
```

Comment:
Take care of using `async void` functions, those should be avoided as much as possible as exceptions thrown inside those functions can't be caught in the outer context, causing the app to crash.
If you really need an `async void` function (usually in event handlers), wrap it with a `try...catch` block.

Reply:
Yes, I had this problem with another project... I added the try...catch block for all the commands.